src/main/java/uk/gov/ons/ctp/integration/mockcaseapiservice/endpoint/CaseServiceMockStub.java

```diff
@@ -120,25 +124,13 @@
       throw new IllegalStateException("Can't supply individualCaseId if not for an individual");
     }
 
-    String caseType = caseDetails.getCaseType();
-    if (caseType.equals("CE") && individual == false) {
-      log.info("Generating new questionnaire ID for CE");
-    } else if (caseType.equals("CE") && individual == true && individualCaseId == null) {
-      log.info("Generating new questionnaire ID for individual in CE");
-    } else if (caseType.equals("HH") && individual == false) {
-      log.info("Generating new questionnaire ID for HH");
-    } else if (caseType.equals("HH") && individual == true && individualCaseId != null) {
-      log.info("Generating new questionaire ID for individual in HH");
-    } else {
-      throw new IllegalStateException(
-          "Invalid combination of caseType, individual and individualCaseId");
-    }
+    RequestValidator.validateGetNewQidByCaseIdRequest(caseDetails, individual, individualCaseId);
 
     SingleUseQuestionnaireIdDTO newQuestionnaire = new SingleUseQuestionnaireIdDTO();
     newQuestionnaire.setQuestionnaireId(
         String.format("%010d", new Random().nextInt(Integer.MAX_VALUE)));
-    newQuestionnaire.setUac("bk5pkrx5hscrclb7");
-    newQuestionnaire.setFormType(formType(caseType).name());
+    newQuestionnaire.setUac(RandomStringUtils.randomAlphanumeric(UAC_LENGTH));
```

Comment:
In future we might want to use a more realistic UAC generator, such as the one in RH-Cuc.